algorithms/linfa-lars/src/algorithm.rs

```diff
@@ -0,0 +1,646 @@
+    Float,
+    dataset::{AsSingleTargets, WithLapack, WithoutLapack},
+    prelude::Records,
+    traits::Fit,
+};
+
+#[cfg(not(feature = "blas"))]
+use linfa_linalg::triangular::{SolveTriangularInplace, UPLO};
+use ndarray::{
+    Array, Array1, Array2, ArrayBase, ArrayView, ArrayView1, ArrayView2, Axis, CowArray, Data,
+    Dimension, Ix2, NewAxis, RemoveAxis, s,
+};
+#[cfg(feature = "blas")]
+use ndarray_linalg::{Diag, Lapack, SolveTriangularInplace, UPLO, layout::MatrixLayout};
+use ndarray_stats::QuantileExt;
+
+use crate::{Lars, LarsValidParams, error::LarsError};
+
+impl<F, D, T> Fit<ArrayBase<D, Ix2>, T, LarsError> for LarsValidParams<F>
+where
+    T: AsSingleTargets<Elem = F>,
+    D: Data<Elem = F>,
+    F: Float,
+{
+    type Object = Lars<F>;
+    /// Fit an LARS model given a feature matrix `x` and a target variable `y`.
+    ///
+    /// The feature matrix `x` must have shape `(n_samples, n_features)`
+    /// The target variable `y` must have shape `(n_samples)`
+    ///
+    /// Returns a `FittedLARS` object which contains the fitted
+    /// parameters and can be used to `predict` values of the target variable
+    /// for new feature values.
+    fn fit(
+        &self,
+        dataset: &linfa::DatasetBase<ArrayBase<D, Ix2>, T>,
+    ) -> Result<Self::Object, LarsError> {
+        let targets = dataset.as_single_targets();
+        let (intercept, y) = compute_intercept(self.fit_intercept(), targets);
+
+        let (alphas, active, coef_path, n_iter) = lars_path(
+            &dataset.records().view(),
+            &y.view(),
+            self.n_nonzero_coefs(),
+            self.eps(),
+            self.verbose(),
+            F::zero(),
+        );
+
+        let intercept = intercept.into_scalar();
+
+        let hyperplane = coef_path.slice(s![.., -1]).to_owned();
+
+        Ok(Lars {
+            hyperplane,
+            intercept,
+            alphas,
+            n_iter,
+            active,
+            coef_path,
+        })
+    }
+}
+
+/// Compute the intercept as the mean of `y` along each column and center `y` if an intercept
+/// should be used, use 0 as intercept and leave `y` unchanged otherwise.
+/// If `y` is 2D, mean is 1D and center is 2D. If `y` is 1D, mean is a number and center is 1D.
+fn compute_intercept<F: Float, I: RemoveAxis>(
+    with_intercept: bool,
+    y: ArrayView<F, I>,
+) -> (Array<F, I::Smaller>, CowArray<F, I>)
+where
+    I::Smaller: Dimension<Larger = I>,
+{
+    if with_intercept {
+        let y_mean = y
+            // Take the mean of each column (1D array counts as 1 column)
+            .mean_axis(Axis(0))
+            .expect("Axis 0 length of 0");
+        // Subtract y_mean from each "row" of y
+        let y_centered = &y - &y_mean.view().insert_axis(Axis(0));
+        (y_mean, y_centered.into())
+    } else {
+        (Array::zeros(y.raw_dim().remove_axis(Axis(0))), y.into())
+    }
+}
+
+/// Compute Least Angle Regression using LARS algorithm
+///
+/// References
```

Comment:
I suppose you have ported the [scikit-learn implementation](https://github.com/scikit-learn/scikit-learn/blob/1eb422d6c5f46a98a318f341de3e4709f9521bfe/sklearn/linear_model/_least_angle.py#L411) maybe you can reference it here and move the references to the lib documentation